src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentTrainEnumerator.cs

```diff
@@ -41,56 +55,86 @@ public ReadOnlySpan<byte> AnalyzeSingleTerm(int fieldId, ReadOnlySpan<byte> valu
 
         public void WriteNull(int fieldId, string path)
         {
-            _allocator.From(global::Corax.Constants.NullValueSlice.AsSpan(), out var b);
-            _terms.Add((fieldId, path, b));
+            // nothing to do here
         }
 
         public void Write(int fieldId, ReadOnlySpan<byte> value)
         {
             if (value.Length == 0)
                 return;
-            _allocator.From(value, out var b);
-            _terms.Add((fieldId, null, b));
+
+            if (_mapping.TryGetByFieldId(fieldId, out var field) == false)
+                return;
+
+            // var analyzer = field.Analyzer ?? _lowercaseAnalyzer;
```

Comment:
fairly certain that we need analyzer.

`"Name": "HIBERNATING RHINOS"`

We'll index that as `"hibernating rhinos"`, and won't benefit from the compression

Reply:
He removed that part intentionally - only for testing: https://issues.hibernatingrhinos.com/issue/RavenDB-21049#focus=Comments-67-811191.0-0